### ISSUE TITLE
Crash: Toast.makeText() in Activities

### ISSUE THREAD

**Issue by endre-borcsok** [nextcloud/android#5504]
### Steps to reproduce
1. Disable WiFi and Mobile data
2. Open Activities tab
3. Bang!

Extra: Unable to recreate in the current PlayStore version

### Expected behaviour
- Toast text should appear

### Actual behaviour
- App crash

### More info

https://stackoverflow.com/questions/48152659/toast-maketext-giving-error-inflating-class-textview-exception

https://issuetracker.google.com/issues/64053344

### Environment data
Android version: 10

Device model: Pixel 2

Stock or customized system: Stock

Nextcloud app version: 3.11.0-alpha1

Nextcloud server version: Unkwon

### Logs
```
2020-02-22 15:04:48.211 27840-27840/com.nextcloud.client E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.nextcloud.client, PID: 27840
    android.view.InflateException: Binary XML file line #27 in android:layout/transient_notification: Binary XML file line #27 in android:layout/transient_notification: Error inflating class android.widget.TextView
    Caused by: android.view.InflateException: Binary XML file line #27 in android:layout/transient_notification: Error inflating class android.widget.TextView
    Caused by: java.lang.reflect.InvocationTargetException
        at java.lang.reflect.Constructor.newInstance0(Native Method)
        at java.lang.reflect.Constructor.newInstance(Constructor.java:343)
        at android.view.LayoutInflater.createView(LayoutInflater.java:854)
        at android.view.LayoutInflater.createView(LayoutInflater.java:776)
        at com.android.internal.policy.PhoneLayoutInflater.onCreateView(PhoneLayoutInflater.java:58)
        at android.view.LayoutInflater.onCreateView(LayoutInflater.java:930)
        at android.view.LayoutInflater.onCreateView(LayoutInflater.java:950)
        at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:1004)
        at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:961)
        at android.view.LayoutInflater.rInflate(LayoutInflater.java:1123)
        at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:1084)
        at android.view.LayoutInflater.inflate(LayoutInflater.java:682)
        at android.view.LayoutInflater.inflate(LayoutInflater.java:534)
        at android.view.LayoutInflater.inflate(LayoutInflater.java:481)
        at android.widget.Toast.makeText(Toast.java:290)
        at android.widget.Toast.makeText(Toast.java:276)
        at com.owncloud.android.ui.activities.ActivitiesActivity.showActivitiesLoadError(ActivitiesActivity.java:263)
        at com.owncloud.android.ui.activities.ActivitiesPresenter$1.onActivitiesLoadedError(ActivitiesPresenter.java:66)
        at com.owncloud.android.ui.activities.data.activities.RemoteActivitiesRepository$1.onError(RemoteActivitiesRepository.java:47)
        at com.owncloud.android.ui.activities.data.activities.ActivitiesServiceApiImpl$GetActivityListTask.onPostExecute(ActivitiesServiceApiImpl.java:143)
        at com.owncloud.android.ui.activities.data.activities.ActivitiesServiceApiImpl$GetActivityListTask.onPostExecute(ActivitiesServiceApiImpl.java:67)
        at android.os.AsyncTask.finish(AsyncTask.java:755)
        at android.os.AsyncTask.access$900(AsyncTask.java:192)
        at android.os.AsyncTask$InternalHandler.handleMessage(AsyncTask.java:772)
        at android.os.Handler.dispatchMessage(Handler.java:107)
        at android.os.Looper.loop(Looper.java:214)
        at android.app.ActivityThread.main(ActivityThread.java:7356)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930)
     Caused by: java.lang.UnsupportedOperationException: Failed to resolve attribute at index 6: TypedValue{t=0x2/d=0x101009b a=1}
        at android.content.res.TypedArray.getColorStateList(TypedArray.java:596)
        at android.widget.TextView.readTextAppearance(TextView.java:3972)
        at android.widget.TextView.<init>(TextView.java:1032)
        at android.widget.TextView.<init>(TextView.java:970)
        at android.widget.TextView.<init>(TextView.java:966)
        at java.lang.reflect.Constructor.newInstance0(Native Method) 
        at java.lang.reflect.Constructor.newInstance(Constructor.java:343) 
        at android.view.LayoutInflater.createView(LayoutInflater.java:854) 
        at android.view.LayoutInflater.createView(LayoutInflater.java:776) 
        at com.android.internal.policy.PhoneLayoutInflater.onCreateView(PhoneLayoutInflater.java:58) 
        at android.view.LayoutInflater.onCreateView(LayoutInflater.java:930) 
        at android.view.LayoutInflater.onCreateView(LayoutInflater.java:950) 
        at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:1004) 
        at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:961) 
        at android.view.LayoutInflater.rInflate(LayoutInflater.java:1123) 
        at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:1084) 
        at android.view.LayoutInflater.inflate(LayoutInflater.java:682) 
        at android.view.LayoutInflater.inflate(LayoutInflater.java:534) 
        at android.view.LayoutInflater.inflate(LayoutInflater.java:481) 
        at android.widget.Toast.makeText(Toast.java:290) 
        at android.widget.Toast.makeText(Toast.java:276) 
        at com.owncloud.android.ui.activities.ActivitiesActivity.showActivitiesLoadError(ActivitiesActivity.java:263) 
        at com.owncloud.android.ui.activities.ActivitiesPresenter$1.onActivitiesLoadedError(ActivitiesPresenter.java:66) 
        at com.owncloud.android.ui.activities.data.activities.RemoteActivitiesRepository$1.onError(RemoteActivitiesRepository.java:47) 
        at com.owncloud.android.ui.activities.data.activities.ActivitiesServiceApiImpl$GetActivityListTask.onPostExecute(ActivitiesServiceApiImpl.java:143) 
        at com.owncloud.android.ui.activities.data.activities.ActivitiesServiceApiImpl$GetActivityListTask.onPostExecute(ActivitiesServiceApiImpl.java:67) 
        at android.os.AsyncTask.finish(AsyncTask.java:755) 
        at android.os.AsyncTask.access$900(AsyncTask.java:192) 
        at android.os.AsyncTask$InternalHandler.handleMessage(AsyncTask.java:772) 
        at android.os.Handler.dispatchMessage(Handler.java:107) 
        at android.os.Looper.loop(Looper.java:214) 
        at android.app.ActivityThread.main(ActivityThread.java:7356) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930) 
```

**Comment by endre-borcsok** [comment]
Suggested fix: replace Toast.makeText() to: DisplayUtils.showSnackMessage(this, error);

Some work has been done ages ago here: https://github.com/nextcloud/android/pull/2389

Seems lik Toast.makeText() is invoked 23 times in the current codebase. I'm happy to raplce them for you if you think that's safe to do so.

**Comment by AndyScherzinger** [comment]
> Seems like Toast.makeText() is invoked 23 times in the current codebase. I'm happy to raplce them for you if you think that's safe to do so.

@tobiasKaminsky what do you think? I'd say @endre-borcsok's suggestion should work while I can't say if simply replacing all occurrences works out fine while I should definitely work in this specific scenario for activities.

**Comment by ezaquarii** [comment]
I guess any activity using this helper must satisfy some extra layout contract, right? I remember that toast did not have any requirements and I'm concerned that we'll start patching thing for edge cases.

I'd consider a mixin approach instead of a static helper, similar to our session mixin.

We could swap the pop-up mechanism depending on needs, get access to currebt ui state (undo with timeout), have better control over ui lifecycle and simplify test assertions.

I won't be hardliner on this. Just a suggestion.

**Comment by endre-borcsok** [comment]
I think the crash was introduced in the 'implementation androidx.appcompat:appcompat:1.2.0-alpha02'
dependecy. Reverting it to the previously used 'appcompat:1.1.0' fixes the crash too.

At the moment any Toast.makeText() would cause a crash on my phone (however, I only tested the ones in Activities). Chances are Google fix this in the production ready '1.2.0'.

For now, I just replace the Toasts in Actvities to DisplayUtils as that seems like a meaningful change anyway. PR is on the way.
